export/kafka.go

```diff
@@ -0,0 +1,200 @@
+package export
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/pkg/errors"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	messagesSent    = stats.Int64("messages_sent", "the number of metric messages successfully sent", "1")
+	messagesDropped = stats.Int64("messages_dropped", "the number of metric messages dropped", "1")
+)
+
+var (
+	messagesSentView = &view.View{
+		Name:        "messages_sent",
+		Measure:     messagesSent,
+		Description: "the number of metric messages successfully sent",
+		Aggregation: view.Count(),
+	}
+
+	messagesDropedView = &view.View{
+		Name:        "messages_dropped",
+		Measure:     messagesDropped,
+		Description: "the number of metric messages dropped",
+		Aggregation: view.Sum(),
+	}
+)
+
+// TopicConfig holds the configurations for Topic info
+type TopicConfig struct {
+	Topic         string
+	NumPartitions int
+	NumReplicas   int
+}
+
+// Kafka is an exporter that exports metrics to a
+// Kafka broker.
+type Kafka struct {
+	config              Config
+	kafkaConfig         *kafka.ConfigMap
+	producer            *kafka.Producer
+	topicInfo           TopicConfig
+	messageFlushTimeSec int
+	lastDroppedLogCount int
+}
+
+// NewKafka returns a new Kafka exporter
+func NewKafka(config Config, kafkaConfig *kafka.ConfigMap, topicInfo TopicConfig) (*ExporterAgent, error) {
+	if err := view.Register(messagesSentView, messagesDropedView); err != nil {
+		return nil, errors.Wrap(err, "Error registering views")
+	}
+
+	createTopic(topicInfo, kafkaConfig)
```

Comment:
User may have/want to create the topic ahead by using cli, or c3 or etc, instead of in the code. We should also make createTopic as an optional choice. Provides the option of configuring which topic to persist to, instead of have to create the topic every time when we start the kafka exporter service.

Reply:
that createTopic function only creates a topic if it doesn't already exist, I've added a comment to clarify this